src/store/state/utils.ts

```diff
@@ -18,3 +18,18 @@ export const generateBudgetYear = (year: number): BudgetYear => ({
   income: [generateBudgetGroup('General', ['Other'])],
   expenses: [generateBudgetGroup('General', ['Other'])]
 });
+
+export const generateBudgetYearFromCurrent = (year: number, current: BudgetYear): BudgetYear => {
+  function clearValues(value: BudgetGroup) {
+    for (const budget of value.budgets) {
+      budget.values = new Array(12).fill(0);
+    }
+    return value;
+  }
```

Comment:
Sorry, I just noticed that this as well overrides the income and expenses budget groups, you'll need to generate new `BudgetGroups`. Make sure to just take the budget group names, do not modify contained values as this will modify the values inside of the budget group from the current year.

> [!NOTE]
> You can try this yourself by filling out the income and expenses for e.g. 2024 and switching to 2025, you'll notice that now all values in 2024 are cleared.

Reply:
Good catch! I think I've fixed it now.